src/storage/iceberg_type.cpp

```diff
@@ -0,0 +1,128 @@
+#include "rest_catalog/objects/map_type.hpp"
+#include "rest_catalog/objects/struct_type.hpp"
+#include "rest_catalog/objects/struct_field.hpp"
+#include "rest_catalog/objects/type.hpp"
+
+namespace duckdb {
+
+string IcebergTypeHelper::LogicalTypeToIcebergType(LogicalType &type) {
+	switch (type.id()) {
+	case LogicalTypeId::TINYINT:
+	case LogicalTypeId::UTINYINT:
+	case LogicalTypeId::SMALLINT:
+	case LogicalTypeId::USMALLINT:
+	case LogicalTypeId::INTEGER:
+		return "int";
+	case LogicalTypeId::BOOLEAN:
+		return "boolean";
+	case LogicalTypeId::VARCHAR:
+		return "string";
+	case LogicalTypeId::DATE:
+		return "date";
+	case LogicalTypeId::UINTEGER:
+	case LogicalTypeId::BIGINT:
+		return "long";
+	case LogicalTypeId::FLOAT:
+		return "float";
+	case LogicalTypeId::DOUBLE:
+		return "double";
+	case LogicalTypeId::DECIMAL: {
+		auto aux = type.AuxInfo()->Cast<DecimalTypeInfo>();
+		return StringUtil::Format("decimal(%d, %d)", aux.width, aux.scale);
+	}
+	case LogicalTypeId::UUID:
+		return "uuid";
+	case LogicalTypeId::BLOB:
+		return "binary";
+	case LogicalTypeId::STRUCT:
+		return "struct";
+	case LogicalTypeId::ARRAY: {
+		// Iceberg doesn't support fixed array lengths
+		return "list";
+	}
+	case LogicalTypeId::LIST:
+		return "list";
+	case LogicalTypeId::TIME:
+		return "time";
+	case LogicalTypeId::TIMESTAMP:
+		return "timestamp";
+	case LogicalTypeId::TIMESTAMP_TZ:
+		return "timestamptz";
+
+	case LogicalTypeId::MAP:
+		return "map";
+	default:
+		throw NotImplementedException("Type %s not supported in Iceberg", LogicalTypeIdToString(type.id()));
+	}
+}
+
+rest_api_objects::Type IcebergTypeHelper::CreateIcebergRestType(LogicalType &type, std::function<idx_t()> get_next_id) {
+	rest_api_objects::Type rest_type;
+
+	switch (type.id()) {
+	case LogicalTypeId::MAP: {
+		rest_type.has_primitive_type = false;
+		rest_type.has_map_type = true;
+		rest_type.map_type = rest_api_objects::MapType();
+		auto aux = type.AuxInfo()->Cast<ListTypeInfo>();
+		auto child_type = aux.child_type.AuxInfo()->Cast<StructTypeInfo>();
+		// How do I get the child types for a map type?
+		// maybe both should just be string?
+		auto key_type = child_type.child_types.front().second;
+		rest_type.map_type.key_id = static_cast<int32_t>(get_next_id());
+		rest_type.map_type.key = make_uniq<rest_api_objects::Type>(
+		    IcebergTypeHelper::CreateIcebergRestType(child_type.child_types.front().second, get_next_id));
+		rest_type.map_type.value_id = static_cast<int32_t>(get_next_id());
+		rest_type.map_type.value = make_uniq<rest_api_objects::Type>(
+		    IcebergTypeHelper::CreateIcebergRestType(child_type.child_types.back().second, get_next_id));
+		rest_type.map_type.value_required = false;
+		return rest_type;
+	}
+	case LogicalTypeId::STRUCT: {
+		rest_type.has_primitive_type = false;
+		rest_type.has_struct_type = true;
+		rest_type.struct_type = rest_api_objects::StructType();
+		auto stuct_aux = type.AuxInfo()->Cast<StructTypeInfo>();
```

Comment:
It's never required to access the aux directly, we have helper methods
`auto &children = StructType::GetChildTypes(type);`